### PR TITLE
Remediation sweep: harden guide validation and align docs

### PR DESCRIPTION
**Pull request by plx**
## Summary
- implement all remediation tasks from the due-diligence review (P1 through P3)
- harden CLI error reporting, guide block parsing, and root-boundary containment
- align syntax/docs policy, expand path expressiveness, and define UTF-8 scope
- improve placeholder/comment semantics, ignore-attribute parsing, and regression coverage
- perform low-risk cleanup (dead surface/deps and validator recursion)

## Validation
- cargo fmt -- --check
- cargo clippy -- -D warnings
- cargo test

## Tracking
All remediation plans are now marked as complete under `remediations/` with `DONE-` prefixes.

### CODE REVIEW

**Comment by plx** [comment]
Follow-up on review feedback from conversation comment IC_kwDOPVHTq87o9spx:\n\nI implemented all code-impacting suggestions in this PR update. Two recommendations are intentionally deferred with no code change in this PR:\n\n1. TOCTOU README note (follow-up item #6): no functional bug was identified here. Root containment checks are already enforced via canonicalization during verification; adding a TOCTOU caveat is documentation polish, not a merge blocker.\n2. Large-guide benchmark (follow-up item #7): useful, but this is performance instrumentation rather than a correctness fix. There is no performance regression signal in the current test suite, so this is deferred to a separate performance-focused change.\n\nNo further change is required for these two items to maintain correctness for this remediation PR.

File: src/validator.rs [hunk 1]
```diff
@@ -95,52 +96,45 @@ impl Validator {
...
+        if path_obj.is_absolute() {
+            return Err(SyntaxError::InvalidPathFormat {
+                line: item.line_number,
+                path: path.to_string(),
+            }
+            .into());
+        }
+
+        // Validate raw slash-separated components so we preserve `.` / `..` and empty components.
+        for component in path.split('/') {
```
[the remark this answers is not in the document]
Comment (plx) [reply src/validator.rs]: Fixed in ae95d1e by validating dot and parent components across both slash and backslash separators, with explicit tests for backslash paths.

**Comment by plx** [comment]
Follow-up on Claude feedback from https://github.com/plx/agentic-navigation-guide/pull/8#issuecomment-3910878445:

Implemented in cc98b46:
1) `resolve_for_containment` now includes an explicit parent-progress guard while walking missing ancestors, so the containment probe cannot spin if parent traversal fails to progress.
2) Added targeted parser tests for the exact ignore-attribute edge cases called out:
   - spaces around `=` (`ignore = "true"`)
   - duplicate keys (`ignore=false ignore=true`)
   - malformed quoted value (`ignore="tr"ue"`)

Also note: this branch already includes tests for circular symlink handling and malformed attributes from prior remediation commits.

Local verification after this change:
- `cargo fmt -- --check`
- `cargo clippy -- -D warnings`
- `cargo test`
all pass.

**Comment by plx** [comment]
Follow-up on the latest review feedback (notably https://github.com/plx/agentic-navigation-guide/pull/8#issuecomment-3910924621):

Implemented in `9f54d33`:
1. Fixed the blocking `--force` mismatch in `init`:
   - `src/cli/init.rs`: removed the nonexistent `--force` guidance from the existing-output error message.
   - `tests/cli_tests.rs`: added a regression assertion that the stderr no longer mentions `--force`.
2. Took two additional low-risk review suggestions:
   - `src/verifier.rs`: added an explicit comment documenting why containment checks do not follow the final symlink component for symlink guide entries.
   - `README.md`: added a concise root-boundary bullet clarifying that paths resolving outside the guide root (including symlink escapes) are rejected.

Local validation on this commit:
- `cargo fmt -- --check`
- `cargo clippy -- -D warnings`
- `cargo test`
all pass.

Remaining items from that comment (resource limits, broader performance tuning, additional optional stress/perf tests) are valid follow-up enhancements but are intentionally not part of this remediation PR scope.
